be/src/service/service_be/starrocks_be.cpp

```diff
@@ -85,14 +85,16 @@ Status init_datacache(GlobalEnv* global_env, const std::vector<StorePath>& stora
         }
         cache_options.mem_space_size = parse_conf_datacache_mem_size(config::datacache_mem_size, mem_limit);
         if (config::datacache_disk_path.value().empty()) {
-            // If the disk cache does not be configured for datacache, set default path according storage path.
+            // If the disk cache does not be configured for datacache, set default path according storage path,
+            // and turn on the automatic adjust switch.
```

Comment:
not sure if this is a good idea, especially when it is the the shared-nothing arch, where the storage paths are usually planned for the tablet storage, user may or may not be aware that the cache will occupy the disk space as well.

Reply:
> not sure if this is a good idea, especially when it is the the shared-nothing arch, where the storage paths are usually planned for the tablet storage, user may or may not be aware that the cache will occupy the disk space as well.

Yes, so we store cache data in other directories, which are not subdirectories of the storage paths, to avoid pollute the storage content. For the disk spaces, we use the disk monitor to automatically adjust cache quota based on current disk usage.

Reply:
it is disk usage that matters, usually the user will not care about what has been written to the directory, but cares about what occupies the disk, when the number of `show data` doesn't match the disk usage.